### PR TITLE
cmake: cleanups about systemd and install

### PR DESCRIPTION
**Pull request by huww98**
This is part of #40844. But that one does not make progress in some time, so I want to at least merge in these cleanups first to avoid some future merge conflicts.

Fixes: https://tracker.ceph.com/issues/53313

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests
- Teuthology
  - [ ] Completed teuthology run
  - [ ] No teuthology test necessary (e.g., documentation)

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by huww98** [comment]
Which tracker checklist item should I select for such refactor PR?

**Comment by ideepika** [comment]
> Which tracker checklist item should I select for such refactor PR?

@huww98 you can create a tracker ticket for this task and add it as reference here.

File: systemd/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,14 @@
-set(CMAKE_INSTALL_SYSTEMD_SERVICEDIR "${CMAKE_INSTALL_LIBEXECDIR}/systemd/system"
```
Comment (tchaikov) [inline systemd/CMakeLists.txt:1]: good point. we should not name this variable with `CMAKE`. as it is not defined by CMake, but by ourselves.

**Review by tchaikov — APPROVED** [review]

**Comment by huww98** [comment]
ping

**Comment by tchaikov** [comment]
@yuriw could you please include this changeset in your next batch?

**Comment by huww98** [comment]
@tchaikov @yuriw Happy new year. Could you consider this PR?

**Comment by ljflores** [comment]
Failures, unrealted:
    https://tracker.ceph.com/issues/53789
    https://tracker.ceph.com/issues/53422
    https://tracker.ceph.com/issues/50192
    https://tracker.ceph.com/issues/53807
    https://tracker.ceph.com/issues/53424
    
Details:
    Bug_#53789: CommandFailedError (rados/test_python.sh): "RADOS object not found" causes test_rados.TestWatchNotify.test_aio_notify to fail - Ceph - RADOS
    Bug_#53422: tasks.cephfs.test_nfs.TestNFS.test_export_create_with_non_existing_fsname: AssertionError: NFS Ganesha cluster deployment failed - Ceph - Orchestrator
    Bug_#50192: FAILED ceph_assert(attrs || !recovery_state.get_pg_log().get_missing().is_missing(soid) || (it_objects != recovery_state.get_pg_log().get_log().objects.end() && it_objects->second->op == pg_log_entry_t::LOST_REVERT)) - Ceph - RADOS
    Bug_#53807: Hidden ansible output and offline filesystem failures lead to dead jobs - Ceph - CephFS
    Bug_#53424: CEPHADM_DAEMON_PLACE_FAIL in orch:cephadm/mgr-nfs-upgrade/ - Ceph - Orchestrator

**Pull request merged**